metrics-operator/controllers/common/providers/provider.go

```diff
@@ -23,7 +23,7 @@ type KeptnSLIProvider interface {
 	FetchAnalysisValue(ctx context.Context, query string, spec metricsapi.AnalysisSpec, provider *metricsapi.KeptnMetricsProvider) (string, error)
 }
 
-type ProviderFactory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
+type NewProviderFactory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
```

Comment:
```suggestion
type NewFactory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
```
or directly
```suggestion
type Factory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
```
This way it reads better in the usage places: `providers.Factory`

Some reading about pkg name in Go: https://go.dev/blog/package-names

Reply:
adapted